### ISSUE TITLE
Shopify translation not working

### ISSUE THREAD

**Issue by danielbixel** [vercel/commerce#314]
In order to get the Shopify product translations, we have to add the Accept-Language Header to the API request.
I did this in the fetch-graphql-api.ts file.

`const res = await fetch(API_URL, {
    ...fetchOptions,
    method: 'POST',
    headers: {
      'Accept-Language': 'en-US',
      'X-Shopify-Storefront-Access-Token': API_TOKEN!,
      ...fetchOptions?.headers,
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      query,
      variables,
    }),
  })`

I can hardcode the language (en-US or es) and it works well. I am getting all my translations.
But how can I get this dynamically connected with the I18nWidget?
Out of the box, the Accept-Language Header is not set and the I18nWidget does not work.

I tried to get the current local from the useRouter hook but I am not able to use the hook in the fetch-graphql-api.ts file

`
const fetchGraphqlApi: GraphQLFetcher = async (
  query: string,
  { variables } = {},
  fetchOptions
) => {
  
  const { locale } = useRouter()
  console.log(locale)

  const res = await fetch(API_URL, {
    ...fetchOptions,
    method: 'POST',
    headers: {
      'Accept-Language': 'en-US',
      'X-Shopify-Storefront-Access-Token': API_TOKEN!,
      ...fetchOptions?.headers,
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      query,
      variables,
    }),
  })
`

But apparently, hooks cannot be used in that file.
Error: Invalid hook call. Hooks can only be called inside of the body of a function component

What is the best way to get Shopify translations working?

Thanks for your help and tips in advance.

**Comment by cond0r** [comment]
@danielbixel https://help.shopify.com/en/manual/cross-border/multilingual-online-store 
Shopify doesn't have this feature out of the box, so translations are not yet implemented into Shopify Provider. But It will be considered in future, following the URL above

**Comment by cond0r** [comment]
@danielbixel yes, the header accept language needs to be sent the at build time for each language supported (like at pages), and also on search page. I will take a look.

**Comment by cond0r** [comment]
@danielbixel what app do you use for translations ? I tried with one, hardcoded header and it didn't sent be back the translated result.

**Comment by danielbixel** [comment]
@cond0r 
The App I used to translate my content manually is called "Translate My Store"
 .. And while writing this I see your response. ;)

This is working perfectly fine.

Thanks a lot
